### PR TITLE
Revert "Revert "start with user creation and login backend""

### PR DESCRIPTION
**Pull request by realswat1**
This is to implement a user creation and login in the backend

### CODE REVIEW

File: expose-app-api/database.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// database.js
+import { Sequelize } from 'sequelize';
+
+export const sequelize = new Sequelize('Exposedatabase', 'exposeswat', 'iamtheSwat1*', {
```
Comment (alfonsocheng) [inline expose-app-api/database.js:4]: non blocking - for capstone, I think this is OK, but in any prod system, you will never commit passwords like this

File: expose-app-api/models/user.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    type: DataTypes.STRING,
+    allowNull: false
+  },
+  created_at: {
+    type: DataTypes.DATE,
+    allowNull: false,
+    defaultValue: sequelize.literal('CURRENT_TIMESTAMP'),
+  },
+
+});
```
Comment (alfonsocheng) [inline expose-app-api/models/user.js:26]: plz add a newline at the end of files so this red symbol goes away XD

File: expose-app-api/routes/user.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        const { username, email, password} = req.body;
+
+        // now checking of user already exists.
+        const existingUser = await user.findOne({where: {email}|| {username}});
+        if (existingUser) {
+            return res.status(400).json({message: 'user already exist'});
+        }
+
+        // Encripting password
+        const hashedPassword = await bcrypt.harsh(password, 10);
```
Comment (alfonsocheng) [inline expose-app-api/routes/user.js:21]: maybe this should be bcrypt.hash

**Comment by alfonsocheng** [comment]
generally we would want to put generated code in a separate PR, or I think in this case, you're missing some .gitignore settings

File: expose-app-api/routes/user.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+const user = require ('../models/users');
+
+
+// user registration 
+router.post('/register', async(req, res)=> { 
+    try {
+
+        const { username, email, password} = req.body;
+
+        // now checking of user already exists.
```
Comment (alfonsocheng) [inline expose-app-api/routes/user.js:14]: I'm not convinced this works, let's revisit when you do the UI

File: expose-app-api/routes/user.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    return res.status(200).json(user);
+    } catch(error){
+        console.error('Error retrieving user profile: ', error);
+        return res.status(500).json({message: 'Internal server error'});
+    }
+
+    // mildware Authentification
+    function authentificationToken(req, res, next){
+        const authHeader = req.Headers['authorization'];
+        const token = authHeader && authHeader.spilt('')[i];
```
Comment (alfonsocheng) [inline expose-app-api/routes/user.js:89]: looks like a typo

File: expose-app-api/database.js (hunk 1 above)
Comment (realswat1) [reply expose-app-api/database.js:4]: how can it be done properly

Comment (alfonsocheng) [reply expose-app-api/database.js:4]: passwords should be stored in some sort of vault storage, it's a whole rabbit hole that we should avoid for now

**Comment by alfonsocheng** [comment]
you can abandon this PR by "closing" it without merging, since all the changes are in #4  now

**Pull request closed without merging**